### ISSUE TITLE
Can't seem to be able to filter warnings in Spacy 3

### ISSUE THREAD

**Issue by davidefiocco** [explosion/spaCy#7033]
## How to reproduce the behaviour

I'm trying to remove stopwords from a sentence, while disabling pipeline steps that (I think!) are unnecessary for the task (and I assume slow down execution!).
Code may read as follows:

```python
import spacy
spacy.cli.download("en_core_web_sm")

# following https://spacy.io/usage/v2-3 on warnings
# I tried also env variables as in https://stackoverflow.com/questions/58294046/ignore-spacy-deprecation-warning
import warnings
warnings.filterwarnings("ignore", message=r"\[W108\]", category=UserWarning)

nlp = spacy.load('en_core_web_sm')
sent = "I want to remove all stopwords from this very sentence."

sent = nlp(sent, disable=['ner', 'parser', 'tagger']) # disabling tagger as I don't think it's needed for stopwords removal?
sent_clean = " ".join([t.text for t in sent if (not t.is_stop)])
```

expected behavior would be to see no `W108` warnings, but I still see them:

```
[W108] The rule-based lemmatizer did not find POS annotation for the token 'I'. Check that your pipeline includes components that assign token.pos, typically 'tagger'+'attribute_ruler' or 'morphologizer'.
[W108] The rule-based lemmatizer did not find POS annotation for the token 'want'. Check that your pipeline includes components that assign token.pos, typically 'tagger'+'attribute_ruler' or 'morphologizer'.
[W108] The rule-based lemmatizer did not find POS annotation for the token 'to'. Check that your pipeline includes components that assign token.pos, typically 'tagger'+'attribute_ruler' or 'morphologizer'.
[W108] The rule-based lemmatizer did not find POS annotation for the token 'remove'. Check that your pipeline includes components that assign token.pos, typically 'tagger'+'attribute_ruler' or 'morphologizer'.
[W108] The rule-based lemmatizer did not find POS annotation for the token 'all'. Check that your pipeline includes components that assign token.pos, typically 'tagger'+'attribute_ruler' or 'morphologizer'.
[W108] The rule-based lemmatizer did not find POS annotation for the token 'stopwords'. Check that your pipeline includes components that assign token.pos, typically 'tagger'+'attribute_ruler' or 'morphologizer'.
[W108] The rule-based lemmatizer did not find POS annotation for the token 'from'. Check that your pipeline includes components that assign token.pos, typically 'tagger'+'attribute_ruler' or 'morphologizer'.
[W108] The rule-based lemmatizer did not find POS annotation for the token 'this'. Check that your pipeline includes components that assign token.pos, typically 'tagger'+'attribute_ruler' or 'morphologizer'.
[W108] The rule-based lemmatizer did not find POS annotation for the token 'very'. Check that your pipeline includes components that assign token.pos, typically 'tagger'+'attribute_ruler' or 'morphologizer'.
[W108] The rule-based lemmatizer did not find POS annotation for the token 'sentence'. Check that your pipeline includes components that assign token.pos, typically 'tagger'+'attribute_ruler' or 'morphologizer'.
[W108] The rule-based lemmatizer did not find POS annotation for the token '.'. Check that your pipeline includes components that assign token.pos, typically 'tagger'+'attribute_ruler' or 'morphologizer'.
```

Am I doing anything silly? Thanks!

## Your Environment

- **spaCy version:** 3.0.1
- **Platform:** Linux-4.19.112+-x86_64-with-Ubuntu-18.04-bionic
- **Python version:** 3.6.9
- **Pipelines:** en_core_web_sm (3.0.0)

**Comment by adrianeboyd** [comment]
To address the underlying cause of the warning:

If you're only removing stop words, you don't need a model like `en_core_web_sm` at all, so you could replace it with:

```python
nlp = spacy.blank("en")
```

Otherwise you could also disable the `lemmatizer` and you also wouldn't have this warning.

----------

To address the use of `filterwarnings`:

The reason `filterwarnings` isn't working is that although the output looks identical, this is a logger warning rather than a `UserWarning`, so you have to manage it with the logger. There are also logger filters, but here I've just set the level to `ERROR` to skip all warnings.

```python
import logging
logger = logging.getLogger("spacy")
logger.setLevel(logging.ERROR)
```

We've been trying to move to using logging more than warnings for warnings/debug information, especially potentially noisy warnings like this one, since it's a lot easier for users to manage and customize, but it's not really consistent at this point and we need to think about how to make what's going on clearer in the output and the docs.

**Comment by davidefiocco** [comment]
Thanks @adrianeboyd for the super prompt reply ! OK, this was indeed was helpful for me as I hadn't realized that I was using `lemmatizer` when it was not needed! My specific case is solved by your suggestions, and I leave it up to you to close the issue or keep it open as a reminder to address noisy warnings with logger/`UserWarning`. Cheers!

**Comment by adrianeboyd** [comment]
Addressed to some degree by #7073.
